docs/cookbook/rag.md

```diff
@@ -0,0 +1,257 @@
+
+```python
+class SoccerPrompt(Prompt):
+    """
+    Here is some context about what happened in soccer in the 2022-23 season:
+    {context}
+
+
+    Based on this information, answer the following question:
+    {question}
+    """
+
+    context: str
+    question: str
+```
+
+But to provide `context`, we need to implement the retrieval functionality. To do this, we embed our question (`query_embedding`), then rank the similarities between `query_embedding` and each of the embeddings from `soccer_info` using the dot product between the vectors. The text with the highest value (and thus, similarity) is passed in as the context. And just like that, our basic RAG example is complete.
+
+```python
+def ask_soccer(query):
+    """Answers a question about soccer from retrieved context."""
+    query_embedding = (
+        chat.client.embeddings.create(model=embeddings_model, input=[query])
+        .data[0]
+        .embedding
+    )
+    # Embedded texts are all the same length, so dot is equivalent to cosine
+    df["similarities"] = df.embeddings.apply(lambda x: np.dot(x, query_embedding))
+    most_similar = df.sort_values("similarities", ascending=False).iloc[0]["texts"]
+
+    prompt = SoccerPrompt(context=most_similar, question=query)
+    completion = chat.create(prompt)
+    print(completion)
+
+
+countries = ["English", "Spanish", "German", "Italian"]
+for country in countries:
+    ask_soccer(f"Who won the {country} top flight?")
+```
+
+## A More Realistic Approach...
+
+!!! note
+
+    This tutorial's code snippets have been rearranged to clearly demonstrate each step of the functionality. For a runnable script, take a look at the [code](https://github.com/Mirascope/mirascope/blob/main/cookbook/basic_rag_example/pinecone_embeddings.py) in our repo. Note that the repo code contains additional functionality and logic flows to make re-runs of the script as efficient as possible.
+
+In reality, we could have passed the entirety of `soccer_info` into the prompt due to its small size. Let's take a look at a practical, scaled up example using the vector database Pinecone to manage the larger data.
+
+Our dataset will be a collection of news articles from 2004 pulled from this [public Github database.](https://raw.githubusercontent.com/Dawit-1621/BBC-News-Classification/main/Data/BBC%20News%20Test.csv) We will pick a topic, then ask our chat model to give a summary of 4 relevant articles.
+
+### Dataset/Preprocessing
+
+Chat and embedding models have token limits on a single call - this wasn't an issue in the soccer example, but these constraints come into play with much larger texts and datasets. For the purposes of this recipe, we split the large texts into roughly equal sizes using helper function `split_text()`. Since we want to be able to fit multiple articles into our final prompt, let's set `max_tokens=1000` and split the text into how many thousands of tokens the original text comprises of.
```

Comment:
explain why 1000 is a good number if we want to fit multiple articles. also that last sentence reads a little weird. Maybe just "split the text into thousand token chunks" would suffice?